File: leetcode/single_number.js
// https://leetcode.com/problems/single-number/
// 136. Single Number
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function (nums) {
    let a = 0;
    for (let i = 0; i < nums.length; i++) {
        a ^= nums[i]
    }
    return a
};
function main() {
    let nums = [1,2,1,3,4,2,4]
    let res = singleNumber(nums)
    console.log(res);
}
main()

File: leetcode/single_number.py
from typing import List
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        n = nums[0]
        for i in range(1, len(nums)):
            n = n ^ nums[i]
        return n

if __name__ == '__main__':
    nums = [1,2,2]
    res = Solution().singleNumber(nums)
    print(res)

